custom-configurations/src/main/java/com/sequenceiq/cloudbreak/domain/CustomConfigurationProperty.java

```diff
@@ -36,11 +45,14 @@
     private String serviceType;
 
     @ManyToOne
+    private Workspace workspace;
```

Comment:
why do you need workspace? It is an old concept, we use account id nowadays. From this change I cannot see the motivation for usage of workspace here

Reply:
In order to store values in the vault, the entity needed to be of type TenantAware or WorkspaceAware. Hence I chose to add workspace as an extra column for the entity

Reply:
you can use `AccountIdAwareResource`, then you won't need to use Workspace or Tenant, only an accountId String

Reply:
several entities in datalake, environment, freeipa service does not have workspace but have secret values, pls check them, for example: `SdxCluster` or `Freeipa` or `DatabaseServer` entities

Reply:
I am sorry, in cloudbreak you need `TenantAwareResource` at least, you are right.

Reply:
but now I would consider to extend SecretAspects in core module to accept `AccountIdAwareResource` too in order to avoid the unnecessary usage of Workspace in new entities.

Reply:
No problem :) In [`SecretAspects.java`](https://github.com/hortonworks/cloudbreak/blob/master/core/src/main/java/com/sequenceiq/cloudbreak/aspect/SecretAspects.java) in the core module throws an exception if the entity is not tenant aware. SecretAspects for the other modules are not written like this.

Reply:
yes, I have checked it minutes ago, the best would if SecretAspects could accept TenantAwareResource and AccountIdAwareResource too, then current logic for old entities in core still would work and also new entities would not have to include workspace and tenant

Reply:
> but now I would consider to extend SecretAspects in core module to accept AccountIdAwareResource too in order to avoid the unnecessary usage of Workspace in new entities.

Okay. There can be two approaches for this that I can think of:
1. `accountId` is already a field in the parent entity: `CustomConfigurations`, but the child entity, `CustomConfigurationProperty`, needs to be made AccountIdAware. Would it make sense to store the child entity within the parent entity as a blob and simply make the parent entity AccountAware? This would involve dropping the child entity altogether
2. Move the accountId field from parent entity to the child entity, and make the child entity AccountAware.

Reply:
You need only to override the getAccountId() method based on AccountIdAwareResource interface, so you do not necessarily need to add accountId field to `CustomConfigurationProperty`, but you can extract it from `customConfigurations` field. What do you think?

Reply:
Okay, yes that is the best approach. I will update this PR with the changes we have discussed. Thanks for your inputs! 😄

Reply:
Updated.